isis/src/mro/apps/hicubenorm/hicubenorm.cpp

```diff
@@ -0,0 +1,760 @@
+    PvlGroup results("Results");
+    for(unsigned int i = 0; i < band.size(); i++) {
+        results += PvlKeyword("Band", toString(band[i]));
+        results += PvlKeyword("RowCol", toString(element[i]));
+        results += PvlKeyword("ValidPixels", toString(validpixels[i]));
+        if(validpixels[i] > 0) {
+        results += PvlKeyword("Mean", toString(average[i]));
+        results += PvlKeyword("Median", toString(median[i]));
+        results += PvlKeyword("Std", toString(stddev[i]));
+        results += PvlKeyword("Minimum", toString(minimum[i]));
+        results += PvlKeyword("Maximum", toString(maximum[i]));
+        }
+        else {
+        results += PvlKeyword("Mean", toString(0.0));
+        results += PvlKeyword("Median", toString(0.0));
+        results += PvlKeyword("Std", toString(0.0));
+        results += PvlKeyword("Minimum", toString(0.0));
+        results += PvlKeyword("Maximum", toString(0.0));
+        }
+    }
+
+    Pvl t;
+    t.addGroup(results);
+    t.write(StatFile);
+    }
+
+    //********************************************************
+    // Create Tabular output of statistics
+    //*******************************************************
+    void tableOut(const QString &StatFile) {
+    // Open output file
+    // TODO check status and throw error
+    ofstream out;
+    out.open(StatFile.toLatin1().data(), std::ios::out);
+
+    // Output a header
+    out << std::setw(8)  << "Band";
+    out << std::setw(8)  << "RowCol";
+    out << std::setw(15) << "ValidPoints";
+    out << std::setw(15) << "Average";
+    out << std::setw(15) << "Median";
+    out << std::setw(15) << "StdDev";
+    out << std::setw(15) << "Minimum";
+    out << std::setw(15) << "Maximum";
+    out << endl;
+
+    // Print out the table results
+    for(unsigned int i = 0; i < band.size(); i++) {
+        out << std::setw(8)  << band[i];
+        out << std::setw(8)  << element[i];
+        out << std::setw(15) << validpixels[i];
+        if(validpixels[i] > 0) {
+        out << std::setw(15) << average[i];
+        out << std::setw(15) << median[i];
+        //Make sure the table's SD is 0 for RowCols with 1 or less valid pixels
+        if(validpixels[i] > 1) {
+            out << std::setw(15) << stddev[i];
+        }
+        else {
+            out << std::setw(15) << 0;
+        }
+        out << std::setw(15) << minimum[i];
+        out << std::setw(15) << maximum[i];
+        }
+        else {
+        out << std::setw(15) << 0;
+        out << std::setw(15) << 0;
+        out << std::setw(15) << 0;
+        out << std::setw(15) << 0;
+        out << std::setw(15) << 0;
+        }
+        out << endl;
+    }
+    out.close();
+    }
+
+    //********************************************************
+    // Gather statistics from a PVL input file
+    //*******************************************************
+    void PVLIn(const Isis::FileName &filename) {
+    Pvl pvlFileIn;
+    pvlFileIn.read(filename.expanded());
```

Comment:
this was originally filename.name(), which seemed like a bug, so I replaced it with expanded.